plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergFailureRecovery.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg;
+
+import io.trino.testing.AbstractTestFailureRecovery;
+import io.trino.testing.QueryRunner;
+import io.trino.tpch.TpchTable;
+import org.testng.SkipException;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatThrownBy;
+
+public class TestIcebergFailureRecovery
+        extends AbstractTestFailureRecovery
+{
+    @Override
+    protected boolean areWriteRetriesSupported()
+    {
+        return true;
+    }
+
+    @Override
+    protected QueryRunner createQueryRunner(List<TpchTable<?>> requiredTpchTables, Map<String, String> configProperties, Map<String, String> coordinatorProperties)
+            throws Exception
+    {
+        return IcebergQueryRunner.builder()
+                .setInitialTables(requiredTpchTables)
+                .setCoordinatorProperties(coordinatorProperties)
+                .setExtraProperties(configProperties)
+                .build();
+    }
+
+    @Override
+    protected void createPartitionedLineitemTable(String tableName, List<String> columns, String partitionColumn)
+    {
+        // no-op - this is only required by testJoin which is disabled in this class
+    }
+
+    @Override
+    public void testAnalyzeStatistics()
+    {
+        assertThatThrownBy(super::testAnalyzeStatistics)
+                .hasMessageContaining("This connector does not support analyze");
+    }
+
+    @Override
+    public void testDelete()
+    {
+        assertThatThrownBy(super::testDelete)
+                .hasMessageContaining("This connector only supports delete where one or more identity-transformed partitions are deleted entirely");
+    }
+
+    @Override
+    // delete is unsupported for non ACID tables
+    public void testDeleteWithSubquery()
+    {
+        assertThatThrownBy(super::testDelete)
+                .hasMessageContaining("This connector only supports delete where one or more identity-transformed partitions are deleted entirely");
+    }
+
+    @Test(invocationCount = INVOCATION_COUNT)
+    public void testDeletePartitionWithSubquery()
+    {
+        assertThatThrownBy(() -> {
+            testTableModification(
```

Comment:
Yeah - I guess we do not need this test at all here. It would make sense to document behaviour difference for Iceberg if test was part of superclass. 
I will drop for now, but it should be moved to superclass and we should start using `TestingConnectorBehavior` in super class